text/primitive-projections.md

```diff
@@ -0,0 +1,70 @@
+- Title: What model for primitive projections?
+
+----
+
+# Summary
+
+Projections of record fields currently follow different overlapping paths:
+- constructors-based vs destructors-based normal forms
+- `t.(f params)` vs `f params t` concrete representation
+- unfolded `Proj` vs folded `Proj` vs `App` internal representation
+
+This document is an attempt to propose solutions to clarify the picture.
+
+See also [#14640](https://github.com/coq/coq/pull/14640) which tries to set up a context to stabilize a design choice.
+
+## Issue 1: the relative status of `Const`-based projections and `Proj`-based projections
+
+It is common to write `f param t` to mean a compact `Proj` kernel node. It is also useful to have a non compact representation for partial application of a projection or for an unapplied projection. What to do with it?
+
+*Design choice 1*: not a kernel problem: the user (or vernac level) can declare `Definition f' params x := x.(f params)` and use it later as an ordinary definition
+  - Advantages: a simple model
+  - Drawbacks: the name `f'` is different
```

Comment:
I think there is a sub choice of whether the projection names exist in the same namespace as constants, or if they're in different namespaces in which case the name can be reused.

Reply:
This choice is linked to the dedicated projection syntax choice. If you have a syntax `t.(p)` that allows non-primitive projections, whatever the details, then there is an ambiguity.